internal/validate/validate.go

```diff
@@ -0,0 +1,36 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	platformv1alpha1 "github.com/openshift/api/platform/v1alpha1"
+)
+
+// UniquePackage checks that the provided PlatformOperator contains a unique spec.package.name
+// when compared to the other PlatformOperators existing on cluster.
+func UniquePackage(ctx context.Context, cli client.Client, po *platformv1alpha1.PlatformOperator) error {
+	existingPlatformOperators := &platformv1alpha1.PlatformOperatorList{}
+	if err := cli.List(ctx, existingPlatformOperators); err != nil {
+		return err
+	}
+
+	for _, existingPO := range existingPlatformOperators.Items {
```

Comment:
I'd propose the we remove the item from this slice that has the earliest timestamp (and tie-break on UID). That way if we're reconciling that one, it won't be considered a duplicate and we'll be able to keep reconciling it.

Reply:
That makes sense. What about this situation though:

1. A PlatformOperator `x` gets created with package name `a`
2. Another PlatformOperator `y` gets created with package name `b`
3. I update PlatformOperator `x` to use package name `b` to create a duplicate

What should happen in this case? Technically `x` is the earliest timestamp so by this logic it would reconcile properly.